src/utils.js

```diff
@@ -4,3 +4,11 @@
 // FIXME: Could not find a way to define a Callback<T> type with multiple
 //        signatures.
 export type Callback = (?Error, any) => void;
+
+export function composeAsync (f1: (...args: Array<*>) => Promise<*>,
+                       f2: (*) => Promise<*>):
+                      (...args: Array<*>) => Promise<*> {
+  return async function composed (...args: Array<*>): Promise<*> {
+    return f2(await f1(...args))
```

Comment:
I'm confused too.

@aenario, @jinroh : maybe one of you can help us understand what happen if `f1` returns a promise that is rejected